partner_credit_limit/models/sale.py

```diff
@@ -1,44 +1,55 @@
+# -*- coding: utf-8 -*-
 # See LICENSE file for full copyright and licensing details.
-
-from datetime import datetime
-
+import logging
 from odoo import api, models, _
 from odoo.exceptions import UserError
-from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+
+_logger = logging.getLogger(__name__)
 
 
 class SaleOrder(models.Model):
+    _name = 'sale.order'
     _inherit = "sale.order"
 
-    @api.multi
     def check_limit(self):
         self.ensure_one()
-        partner = self.partner_id
-        moveline_obj = self.env['account.move.line']
-        movelines = moveline_obj.search(
+        if self.partner_id.commercial_partner_id:
+            partner = self.partner_id.commercial_partner_id
+        else:
+            partner = self.partner_id
+        move_lines = self.env['account.move.line'].search(
             [('partner_id', '=', partner.id),
-             ('account_id.user_type_id.name', 'in', ['Receivable', 'Payable']),
-             ('full_reconcile_id', '=', False)]
+             ('account_id.user_type_id.id', 'in', [
+                 self.env.ref('account.data_account_type_receivable').id,
+                 self.env.ref('account.data_account_type_payable').id]),
+             ('full_reconcile_id', '=', False),
+             ('company_id', '=', self.env.user.company_id.id)]
         )
+
         debit, credit = 0.0, 0.0
-        today_dt = datetime.strftime(datetime.now().date(), DF)
-        for line in movelines:
-            if line.date_maturity < today_dt:
-                credit += line.debit
-                debit += line.credit
+        for line in move_lines:
```

Comment:
Would be better to use `search_read` as we are doing this  for `credit`/`debit` only